transformations/mixed_language_perturbation/transformation.py

```diff
@@ -0,0 +1,89 @@
+import random
+import string
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+"""
+Mixed Language Perturbation
+This perturbation translates words in the text from English to other languages (e.g., German). It can be used to test the robustness of a model in a multilingual setting.
+"""
+
+def translate(model, tokenizer, text, src_lang, target_lang):
+    tokenizer.src_lang = src_lang
+    encoded = tokenizer(text, return_tensors="pt")
+    generated_tokens = model.generate(**encoded, forced_bos_token_id=tokenizer.get_lang_id(target_lang))
+    res = tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+    return res
+
+def mixed_language(model, tokenizer, text, prob_mix=0.1, src_lang="en", trg_lang="fr", seed=0):
+    random.seed(seed)
+
+    words = text.split()
+    mixed_text = ""
+    for word in words:
+        if mixed_text != "":
+            mixed_text += " "
+        if random.random() < prob_mix:
+            plain_word = word.translate(str.maketrans('', '', string.punctuation)).strip()
+
+            if plain_word == "":
+                continue
+
+            mixed_text += translate(model, tokenizer, plain_word, src_lang, trg_lang)[0]
+
+            if word[-1] in string.punctuation:
+                mixed_text += word[-1]
+        else:
+            mixed_text += word
+
+    print(text, mixed_text)
```

Comment:
Maybe it's better to avoid printing inside the transformation because one might want to transform the whole dataset and this can overwhelm the output screen.

Reply:
Thanks for mentioning this! We removed the print.